model_analyzer/config/generate/search_parameters.py

```diff
@@ -153,6 +152,31 @@ def _populate_concurrency(self) -> None:
                 rcs_parameter_max_value=self._config.run_config_search_max_concurrency,
             )
 
+    def _populate_max_batch_size(self) -> None:
+        # Example config format:
+        # model_config_parameters:
+        #  max_batch_size: [1, 4, 16]
+        if not self._model_config_parameters:
+            self._populate_rcs_parameter(
+                parameter_name="max_batch_size",
+                rcs_parameter_min_value=self._config.run_config_search_min_model_batch_size,
+                rcs_parameter_max_value=self._config.run_config_search_max_model_batch_size,
+            )
+        elif "max_batch_size" in self._model_config_parameters.keys():
+            parameter_list = self._model_config_parameters["max_batch_size"]
+
+            self._populate_list_parameter(
+                parameter_name="max_batch_size",
+                parameter_list=parameter_list,
+                parameter_category=ParameterCategory.INT_LIST,
+            )
+        else:
```

Comment:
Is there any difference between the first IF and this final ELSE?

Reply:
No, I've refactored this to make it cleaner.